File: server/mc/1.12.2-cleanroom/src/main/java/calebxzhou/rdi/mc/server/tpa/TpaPlayer112.java
package calebxzhou.rdi.mc.server.tpa;

import calebxzhou.rdi.mc.common2.tpa.TpaPlayer;
import net.minecraft.entity.player.EntityPlayerMP;
import net.minecraft.util.text.TextComponentString;

import java.util.UUID;

public final class TpaPlayer112 implements TpaPlayer {
    private final EntityPlayerMP player;

    public TpaPlayer112(EntityPlayerMP player) {
        this.player = player;
    }

    public EntityPlayerMP unwrap() {
        return player;
    }

    @Override
    public UUID id() {
        return player.getUniqueID();
    }

    @Override
    public String name() {
        return player.getGameProfile().getName();
    }

    @Override
    public void sendMessage(String message) {
        player.sendMessage(new TextComponentString(message));
    }
}

File: server/mc/1.12.2-cleanroom/src/main/kotlin/calebxzhou/rdi/mc/server/tpa/TpaPlayer112.kt
package calebxzhou.rdi.mc.server.tpa

import calebxzhou.rdi.mc.rcmd.tpa.TpaPlayer
import net.minecraft.entity.player.EntityPlayerMP
import net.minecraft.util.text.TextComponentString
import java.util.UUID

class TpaPlayer112(private val player: EntityPlayerMP) : TpaPlayer {
    fun unwrap(): EntityPlayerMP = player

    override fun id(): UUID = player.uniqueID

    override fun name(): String = player.gameProfile.name

    override fun sendMessage(message: String) {
        player.sendMessage(TextComponentString(message))
    }
}
